packages/wallet/src/services/AssetsTracker.ts

```diff
@@ -47,16 +50,23 @@ export const createAssetsTracker = (
         // Fetch asset metadata only for assets not already present in assetsMap
         map((assetIds) =>
           assetIds.map((assetId) => {
-            if (assetsMap.has(assetId)) {
-              return of(assetsMap.get(assetId));
+            const assetInfo = assetsMap.get(assetId);
+            if (assetInfo && isAssetInfoComplete(assetInfo)) {
+              return of(assetInfo);
             }
-            logger.debug(`Fetching metadata for asset ${assetId}`);
-            return assetService(assetId);
+            const stopPolling = new Subject<void>();
+            return intervalBackoff(retryBackoffConfig).pipe(
+              // It starts right away, as opposed to rxjs interval which waits the interval before emitting
+              tap((iter) => logger.debug(`Fetch metadata for asset ${assetId}`, iter)),
+              takeUntil(stopPolling),
+              switchMap(() => assetService(assetId)),
+              tap((v) => isAssetInfoComplete(v) && stopPolling.next())
```

Comment:
I should also emit only distinct values. I will update it but I want to get your feedback before proceeding.

Reply:
obsolete if we go with the `pollUntil` approach in coldObservable. ([347bc46](https://github.com/input-output-hk/cardano-js-sdk/pull/464/commits/347bc467d864e1c109156d7f92a780eadaffc381))